dev/system-test/firestore.ts

```diff
@@ -152,6 +152,24 @@ describe('Firestore class', () => {
         expect(err).to.equal('The client has already been terminated');
       });
   });
+
+  it('throws an error if terminate() is called with active listeners', async () => {
+    const ref = randomCol.doc('doc-1');
+    const unsubscribe = ref.onSnapshot(() => {
+      // No-op
+    });
+
+    await ref.set({});
+    try {
+      await firestore.terminate();
+      throw new Error('terminate() should have failed');
+    } catch (err) {
+      expect(err).to.equal(
+        'All listeners must be disconnected before terminating the client.'
+      );
+      unsubscribe();
+    }
```

Comment:
Nit: If you want, you can simplify this as such: https://github.com/googleapis/nodejs-firestore/blob/7562033876dc006e77d00b576b2541a7dfd30c66/dev/test/transaction.ts#L408